dask/array/tests/test_array_core.py

```diff
@@ -2286,6 +2286,24 @@ def test_astype():
     assert d.astype("f8") is d
 
 
+def test_astype_gh1151():
+    a = np.arange(5).astype(np.int32)
+    b = da.from_array(a, (1,))
+    assert_eq(a.astype(np.int16), b.astype(np.int16))
+
+
+def test_astype_gh9318():
+    # `order`` kwarg in `astype` should not cause an error
+    a = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]], order="C")
+    b = da.from_array(a, chunks=(2, 2))
+    a.astype(float, order="F")
+    b.astype(float, order="F")  # if no error at this line, pytest passes
```

Comment:
Can we add and `assert_eq` call here too? That will test both that `b.astype(float, order="F")` doesn't raise an error (as it does today on `main`) and that `dask` and `numpy` are giving the same result

Reply:
Ok, done. 
I've added a note too - it seems important in this instance to clarify that `assert_eq` does not itself explicitly check the contiguous nature of the arrays.